client/src/features/posts/postsThunks.js

```diff
@@ -11,6 +11,18 @@ const fetchPosts = createAsyncThunk("posts/fetchPosts", async () => {
   const data = await response.json();
   return data;
 });
+const addNewPost = createAsyncThunk('posts/addNewPost', async (newPost) => {
+  const response = await fetch("/posts", {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(newPost),
+  });
+  const data = await response.json();
+  return data;
```

Comment:
At this point, you sent a POST request, the post is sent to the DB, this is great! However, the posts array in our store doesn't know about this, so it does not update despite the fact that a new post was added to our collection. Meaning if you go back to the home page, the post list will not have been updated.

So to let our store know that there was a post added, I think you could dispatch the `addPost` reducer (which already exists in `postsSlice.js`, but we should rename it to `postAdded`) and pass in the new post as a parameter. Like so: `dispatch(postAdded(data));`. This can go right before the return.

Reply:
I am still trying to work out creating the connection. I had a few questions for our next meeting about the connection with the store. I am still trying to figure this out. I am also trying to figure out how to test that things are being stored in the data base as I am working on it, but we can talk more about this in our next meeting.